File: app/src/main/java/com/games/memorygame/model/game/Player.java
package com.themobilecompany.memorygame.model.game;

public class Player {

    private int score;
    private boolean turn;

    public Player(boolean turn) {
        this.score = 0;
        this.turn = turn;
    }

    public int getScore() {
        return score;
    }

    public void addPoints(int points) {
        this.score += points;
    }

    public boolean hasTurn() {
        return this.turn;
    }

    public void changeTurn() {
        this.turn = !this.turn;
    }

}

File: app/src/main/java/com/games/memorygame/model/game/Player.kt
package com.games.memorygame.model.game

class Player(private var turn: Boolean) {

    var score: Int = 0
        private set

    fun addPoints(points: Int) {
        this.score += points
    }

    fun hasTurn(): Boolean = this.turn

    fun changeTurn() {
        this.turn = !this.turn
    }

}

enum class PlayerMode {
    SINGLE_PLAYER,
    MULTI_PLAYER
}
